lib/recurly/coupon.php

```diff
@@ -86,6 +86,18 @@ public function createUpdateXML() {
     return $this->renderXML($doc);
   }
 
+  public function generate($amount) {
+    $doc = $this->createDocument();
+
+    $root = $doc->appendChild($doc->createElement($this->getNodeName()));
+    $root->appendChild($doc->createElement('number_of_unique_codes', $amount));
+
+    $response = $this->_client->request(Recurly_Client::POST, $this->uri() . '/generate', $this->renderXML($doc));
+    $response->assertValidResponse();
+
+    return Recurly_UniqueCouponCodeList::createFromUri($response->headers['Location'], $this->_client);
```

Comment:
What do you think about just returning an array since it's a fixed number?
```php
    $coupons = array();
    foreach (Recurly_UniqueCouponCodeList::createFromUri($response->headers['Location'], $this->_client) as $coupon) {
      $coupons[] = $coupon;
      if (count($coupons) == $amount) break;
    }
    return $coupons;
```